### PR TITLE
Better nutation

### PR DESCRIPTION
**Pull request by guinneak198**

### CODE REVIEW

File: examples/combined_ODNP.py [hunk 1]
```diff
@@ -9,12 +9,11 @@
...
-from pyspecdata import *
-from pyspecdata.file_saving.hdf_save_dict_to_group import (
-    hdf_save_dict_to_group,
-)
+from pyspecdata import init_logging, getDATADIR, figlist_var
+from pyspecdata.file_saving.hdf_save_dict_to_group import hdf_save_dict_to_group
 from pyspecdata import strm
-import os, sys, time
+import os
+import time
```
Comment (guinneak198) [inline examples/combined_ODNP.py]: black

File: SpinCore_pp/ppg/echo.py [hunk 2]
```diff
@@ -96,13 +96,14 @@ def run_spin_echo(
...
     data_length = 2 * nPoints * nEchoes * nPhaseSteps
+    RX_scans = 1
     for nScans_idx in range(nScans):
         run_scans_time_list = [time.time()]
         run_scans_names = ["configure"]
         configureTX(adcOffset, carrierFreq_MHz, tx_phases, amplitude, nPoints)
         run_scans_time_list.append(time.time())
         run_scans_names.append("configure Rx")
-        acq_time_ms = configureRX(SW_kHz, nPoints, nScans, nEchoes, nPhaseSteps)
+        acq_time_ms = configureRX(SW_kHz, nPoints,RX_nScans, nEchoes, nPhaseSteps)
```
Comment (jmfranck) [inline SpinCore_pp/ppg/echo.py]: Typo not same as previous

File: examples/combined_ODNP.py [hunk 3]
```diff
@@ -57,9 +56,7 @@
     if j in config_dict.keys()
 }
 vd_list_us = (
-    SpinCore_pp.vdlist_from_relaxivities(
-        config_dict["concentration"], **vd_kwargs
-    )
+    SpinCore_pp.vdlist_from_relaxivities(config_dict["concentration"], **vd_kwargs)
     * 1e6
```
Comment (guinneak198) [inline examples/combined_ODNP.py]: black

**Review by guinneak198 — COMMENTED** [review]
a lot of diffs after running black and passing flake

File: examples/run_generic_echo.py [hunk 4]
```diff
@@ -9,10 +9,10 @@
 command with 'stayput' (e.g. 'py run_generic_echo.py stayput')
 """
 
-from pylab import *
-from pyspecdata import *
-import os, sys
-from numpy import *
+from pyspecdata import getDATADIR, r_
```
Comment (jmfranck) [inline examples/run_generic_echo.py]: r_ belongs to numpy not pyspecdata

File: examples/run_largecyc_generic_echo.py [hunk 5]
```diff
@@ -136,7 +134,7 @@
     [len(ph1), len(ph_overall), -1],
 )
 data.setaxis("ph1", ph1 / 4).setaxis("ph_overall", ph_overall / 4)
-data.set_prop("postproc_type", "spincore_generalproc_v1")
+data.set_prop("postproc_type", "spincore_generalproc_v3")
```
Comment (jmfranck) [inline examples/run_largecyc_generic_echo.py]: This should be named after the ppg. It should not be called generalproc (this was a mistake I didn't catch before)

File: examples/run_var_tau.py [hunk 6]
```diff
@@ -5,7 +5,7 @@
 A standard echo that is repeated varying the echo time between pulses. The tau value is adjusted 
 to ensure a symmetric echo.
 """
-from pyspecdata import *
+from pyspecdata import figlist_var, r_, getDATADIR
```
Comment (jmfranck) [inline examples/run_var_tau.py]: r_ belongs to numpy

File: examples/run_var_tau.py [hunk 7]
```diff
@@ -35,43 +35,47 @@
...
+    repetition_us=config_dict["repetition_us"],
+    tau_us=tau,
+    SW_kHz=config_dict["SW_kHz"],
+    indirect_fields=("tau_adjust", "tau"),
+    ret_data=None,
 )
 mytau_axis = var_tau_data.getaxis("indirect")
-mytau_axis[0]["tau_adjust"] = tau_adjust
-mytaus_axis[0]["tau"] = tau
+mytau_axis[0]["tau_adjust"] = tau_adjust_range[0]
```
Comment (jmfranck) [inline examples/run_var_tau.py]: Why does this differ??

**Review by jmfranck — CHANGES_REQUESTED** [review]

File: examples/run_nutation_amp_scopecapture.py [hunk 8]
```diff
@@ -0,0 +1,122 @@
+from pyspecdata import *
```
Comment (jmfranck) [inline examples/run_nutation_amp_scopecapture.py]: this file had a bad name, so I renamed it

File: examples/run_var_tau.py (hunk 7 above)
Comment (guinneak198) [reply examples/run_var_tau.py]: tau_adjust does not exist this was a typo I caught as well as mytau_axis vs mytaus_axis

File: examples/combined_ODNP.py [hunk 9]
```diff
@@ -99,8 +97,8 @@
 powers = 1e-3 * 10 ** (dB_settings / 10.0)
 # }}}
 # {{{ these change if we change the way the data is saved
-IR_postproc = "spincore_IR_v1"  # note that you have changed the way the data is saved, and so this should change likewise!!!!
-Ep_postproc = "spincore_ODNP_v3"
+IR_postproc = "spincore_IR_v3"  # note that you have changed the way the data is saved, and so this should change likewise!!!!
+Ep_postproc = "spincore_ODNP_v5"
```
Comment (guinneak198) [inline examples/combined_ODNP.py]: increment post processing version so that we aren't multiplying by nScans anymore

File: examples/run_CPMG.py [hunk 10]
```diff
@@ -159,10 +151,11 @@
...
 )
-data.setaxis("nEcho", r_[0 : config_dict["nEchoes"]]).setaxis(
-    "ph2", ph2 / 4
-).setaxis("ph_diff", ph_diff / 4)
-data.set_prop("postproc_type", "spincore_diffph_SE_v2")
+data.setaxis("nEcho", r_[0 : config_dict["nEchoes"]]).setaxis("ph2", ph2 / 4).setaxis(
+    "ph_diff", ph_diff / 4
+)
+data.set_units("t2", "s")
+data.set_prop("postproc_type", "spincore_CPMG_v2")
```
Comment (guinneak198) [inline examples/run_CPMG.py]: increment post processing version so that we aren't multiplying by nScans anymore

File: examples/run_FS.py [hunk 11]
```diff
@@ -154,7 +153,7 @@
         .set_units("indirect", "scan #")
     )
 sweep_data.name(config_dict["type"] + "_" + str(config_dict["field_counter"]))
-sweep_data.set_prop("postproc_type", "field_sweep_v1")
+sweep_data.set_prop("postproc_type", "field_sweep_v3")
```
Comment (guinneak198) [inline examples/run_FS.py]: increment post processing version so that we aren't multiplying by nScans anymore

File: examples/run_Hahn_echo.py [hunk 12]
```diff
@@ -92,7 +92,8 @@
 )
 data.setaxis("ph1", ph1_cyc / 4)
 data.reorder(["ph1", "nScans", "t2"])
-data.set_prop("postproc_type", "spincore_SE_v1")
+data.set_units("t2", "s")
+data.set_prop("postproc_type", "spincore_SE_v3")
```
Comment (guinneak198) [inline examples/run_Hahn_echo.py]: increment post processing version so that we aren't multiplying by nScans anymore

File: examples/run_Hahn_echo_mw.py [hunk 13]
```diff
@@ -121,38 +120,36 @@
...
     for_plot = echo_data.C
-    for_plot.ft('t2',shift=True)
-    fl.next('FTed data')
+    for_plot.ft("t2", shift=True)
+    fl.next("FTed data")
     fl.image(for_plot)
-echo_data.name(config_dict["type"] + "_" + str(config_dict["echo_counter"])
-echo_data.set_prop("postproc_type", "proc_Hahn_echoph")
+echo_data.name(config_dict["type"] + "_" + str(config_dict["echo_counter"]))
+echo_data.set_prop("postproc_type", "spincore_SE_v3")
```
Comment (guinneak198) [inline examples/run_Hahn_echo_mw.py]: increment post processing version so that we aren't multiplying by nScans anymore

File: examples/run_field_dep_justMw.py [hunk 14]
```diff
@@ -191,7 +186,7 @@
         .set_units("indirect", "scan #")
     )
 sweep_data.name(config_dict["type"] + "_" + str(config_dict["field_counter"]))
-sweep_data.set_prop("postproc_type", "field_sweep_v2")
+sweep_data.set_prop("postproc_type", "field_sweep_v3")
```
Comment (guinneak198) [inline examples/run_field_dep_justMw.py]: increment post processing version so that we aren't multiplying by nScans anymore

File: examples/run_generic_echo.py [hunk 15]
```diff
@@ -131,8 +129,9 @@
     ["ph2", "ph_diff", "t2"],
     [len(ph2), len(ph_diff), -1],
 )
+data.set_units("t2", "s")
 data.setaxis("ph2", ph2 / 4).setaxis("ph_diff", ph_diff / 4)
-data.set_prop("postproc_type", "spincore_diffph_SE_v2")
+data.set_prop("postproc_type", "spincore_diffph_SE_v4")
```
Comment (guinneak198) [inline examples/run_generic_echo.py]: increment post processing version so that we aren't multiplying by nScans anymore

File: examples/run_largecyc_CPMG.py [hunk 16]
```diff
@@ -188,10 +179,11 @@
...
 )
-data.setaxis("nEcho", r_[0 : config_dict["nEchoes"]]).setaxis(
-    "ph1", ph1 / 4
-).setaxis("ph2", ph2 / 4).setaxis("ph_overall", ph_overall / 4)
-data.set_prop("postproc_type", "spincore_generalproc_v1")
+data.set_units("t2", "s")
+data.setaxis("nEcho", r_[0 : config_dict["nEchoes"]]).setaxis("ph1", ph1 / 4).setaxis(
+    "ph2", ph2 / 4
+).setaxis("ph_overall", ph_overall / 4)
+data.set_prop("postproc_type", "spincore_largecyc_CPMG_v1")
```
Comment (guinneak198) [inline examples/run_largecyc_CPMG.py]: increment post processing version so that we aren't multiplying by nScans anymore

Comment (guinneak198) [reply examples/run_largecyc_CPMG.py]: also we don't have a postprocessing function yet for this guy

File: examples/run_largecyc_generic_echo.py [hunk 17]
```diff
@@ -136,7 +134,8 @@
     [len(ph1), len(ph_overall), -1],
 )
 data.setaxis("ph1", ph1 / 4).setaxis("ph_overall", ph_overall / 4)
-data.set_prop("postproc_type", "spincore_generalproc_v1")
+data.set_units("t2", "s")
+data.set_prop("postproc_type", "spincore_largecyc_generic_echo_v1")
```
Comment (guinneak198) [inline examples/run_largecyc_generic_echo.py]: increment post processing version so that we aren't multiplying by nScans anymore and we currently don't have any function to process this guy so v1

File: examples/run_nutation.py [hunk 18]
```diff
@@ -82,16 +82,17 @@
...
+data.chunk("t", ["ph2", "ph1", "t2"], [2, 2, -1])
+data.setaxis("ph1", ph1_cyc / 4).setaxis("ph2", ph2_cyc / 4)
 if config_dict["nScans"] > 1:
     data.setaxis("nScans", r_[0 : config_dict["nScans"]])
-data.reorder(["ph1", "nScans", "t2"])
+data.reorder(["nScans", "ph2", "ph1", "p_90", "t2"])
 data.set_units("t2", "s")
-data.set_prop("postproc_type", "spincore_nutation_v4")
-data.set_prop("coherence_pathway", {"ph1": +1})
+data.set_prop("postproc_type", "spincore_nutation_v5")
```
Comment (guinneak198) [inline examples/run_nutation.py:99]: increment post processing version so that we aren't multiplying by nScans anymore

File: examples/run_nutation_amp_scopecapture.py [hunk 19]
```diff
@@ -0,0 +1,116 @@
+from pyspecdata import figlist_var, getDATADIR, ndshape
+import os
+import SpinCore_pp
+from datetime import datetime
+import numpy as np
+import h5py
+from Instruments.XEPR_eth import xepr
+from Instruments.gds import GDS_scope
+from SpinCore_pp.ppg import run_spin_echo
```
Comment (guinneak198) [inline examples/run_nutation_amp_scopecapture.py]: I don't know how this guy got into this PR should I remove and make a different PR for this? I honestly don't remember what this guy does

Comment (guinneak198) [reply examples/run_nutation_amp_scopecapture.py]: it might be the run_nutation_amplitude.py renamed?

File: examples/run_var_tau.py [hunk 20]
```diff
@@ -139,7 +136,7 @@
     fl.image(for_plot)
 var_tau_data.name(config_dict["type"] + "_" + str(config_dict["echo_counter"]))
 var_tau_data.set_prop(
-    "postproc_type", "SpinCore_var_tau_v1"
+    "postproc_type", "SpinCore_var_tau_v2"
```
Comment (guinneak198) [inline examples/run_var_tau.py]: increment post processing version so that we aren't multiplying by nScans anymore

File: examples/run_nutation_amp_scopecapture.py [hunk 21]
```diff
@@ -0,0 +1,111 @@
+"""
+Amp varied Nutation
```
Comment (jmfranck) [inline examples/run_nutation_amp_scopecapture.py]: This is not an amplitude, very mutation, right?  This was adapted from an earlier example, which is as it should be, but the doc string at the top needs to be edited, and a `git mv` may be in order.

**Review by jmfranck — CHANGES_REQUESTED** [review]
Quick issue noted... Needs further review

File: examples/run_nutation_amp_scopecapture.py (hunk 21 above)
Comment (guinneak198) [reply examples/run_nutation_amp_scopecapture.py]: okay renamed and edited docstring - even in master it was never a nutation just varying the amplitude

File: examples/run_varied_amp_capture.py [hunk 22]
```diff
@@ -0,0 +1,111 @@
+"""
+Amp variation
+=============
+A Hahn echo is output directly to the GDS oscilloscope from the SpinCore.
```
Comment (jmfranck) [inline examples/run_varied_amp_capture.py]: but it should not be a Hahn echo

Comment (jmfranck) [reply examples/run_varied_amp_capture.py]: we are keeping things simple by dealing with one pulse at a time.

Comment (jmfranck) [reply examples/run_varied_amp_capture.py]: have you even tested/used this guy?  You are looping a parameter called "amplitude" that you are not even using.

Comment (jmfranck) [reply examples/run_varied_amp_capture.py]: I think the history here is that I pulled and edited a previous non-working file that is closest to what we wanted to do.  But then you made another file that does what we actually want to do.

I think the solution here is that if you have a file to capture pulses with different pulse lengths on the scope, then just point out what file that is here.  Then you can delete this file, which I think is junk?

**Review by jmfranck — CHANGES_REQUESTED** [review]

File: examples/run_varied_amp_capture.py (hunk 22 above)
Comment (guinneak198) [reply examples/run_varied_amp_capture.py]: I am deleting this because the file you are talking about is in [this PR](https://github.com/jmfrancklab/FLinst/pull/14)

File: examples/run_nutation.py [hunk 23]
```diff
@@ -82,16 +86,18 @@
...
         ret_data=data,
     )
-data.setaxis("indirect", p90_range_us * 1e-6).set_units("indirect", "s")
+data.rename("indirect", "beta")
+data.setaxis("beta", beta_range_s_sqrtW).set_units("beta", "s√W")
+data.set_prop("p_90s", prog_p90_us)
 # {{{ chunk and save data
-data.chunk("t", ["ph1", "t2"], [4, -1])
-data.setaxis("ph1", ph1_cyc / 4)
+data.chunk("t", ["ph2", "ph1", "t2"], [2, 2, -1])
```
Comment (jmfranck) [inline examples/run_nutation.py:93]: Are you sure that this ordering is correct?

**Review by jmfranck — CHANGES_REQUESTED** [review]
Just one double check

File: examples/run_nutation.py (hunk 23 above)
Comment (guinneak198) [reply examples/run_nutation.py:93]: It's the same ordering we have been using for IR and that is chunked correctly

Comment (guinneak198) [reply examples/run_nutation.py:93]: and with testing/processing it shows up in the right pathway
